drivers/adc/adc_mchp_xec.c

```diff
@@ -56,7 +56,7 @@ struct adc_xec_regs {
 	uint32_t single_reg;
 	uint32_t repeat_reg;
 	uint32_t channel_read_reg[XEC_ADC_CFG_CHANNELS];
-	uint32_t unused[10 + (MCHP_ADC_MAX_CHAN - XEC_ADC_CFG_CHANNELS)];
+	uint32_t unused[26 - XEC_ADC_CFG_CHANNELS];
```

Comment:
Can you please update to

`uint32_t unused[10 + (XEC_ADC_MAX_HW_CHAN - XEC_ADC_CFG_CHANNELS)];`

and we can add the macro in this file (preferably above ` XEC_ADC_CFG_CHANNELS` definition)
`#define XEC_ADC_MAX_HW_CHAN 16`

This way the code will be clear to understand

Hardware implements 16 reading registers for all packages. The smaller chip packages only hook are reduced number of ADC inputs. The actual number of channels hooked up to pins from DT will be read from `XEC_ADC_CFG_CHANNELS`